index.bs

```diff
@@ -52,72 +49,205 @@ RTCPeerConnection, canvas or MediaRecorder.
+  MediaStreamTrack signalTarget;
+  MediaStreamTrackGeneratorKind kind;
+}
 </pre>
 
+### Internal slots
+<dl>
+<dt><dfn attribute for=MediaStreamTrackGenerator>[[signalTarget]]</dfn></dt>
+<dd>(Optional) track to which the {{MediaStreamTrackGenerator}} will automatically forward control signals.</dd>
+<dt><dfn attribute for=MediaStreamTrackGenerator>[[kind]]</dfn></dt>
+<dd>Kind of media data for this {{MediaStreamTrackGenerator}}. Must be a valid {{MediaStreamTrackKind}}
+  (i.e., `"audio"`  or `"video"`).</dd>
+</dl>
+
+### Constructor
+<dfn constructor for=MediaStreamTrackGenerator title="MediaStreamTrackGenerator(init)">
+  MediaStreamTrackGenerator(init)
+</dfn>
+1. If {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}} is not empty and is not a valid {{MediaStreamTrack}},
+   or {{init.}}{{MediaStreamTrackGenerator/[kind]}} is not empty and it is not `"audio"` or `"video"`,
+   throw a {{TypeError}}.
+2. If neither {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}} nor
+   and {{init.}}{{MediaStreamTrackGenerator/[kind]}} are empty, and
+   {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}}{{.}}{{MediaStreamTrack/kind}}
+   does not match {{init.}}{{MediaStreamTrackGeneratorInit/kind}}, throw a {{TypeError}}.
+3. Let |g| be a new {{MediaStreamTrackGenerator}} object.
+4. If {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}} is not empty,
+   assign {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}}
+   to the {{MediaStreamTrackGenerator/[signalTarget]}} internal slot of |g| and
+   initialize the {{MediaStreamTrack/kind}} field of |g| (inherited from {{MediaStreamTrack}})
+   with {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}}{{.}}{{MediaStreamTrack/kind}}.
+   Otherwise, initialize the {{MediaStreamTrack/kind}} field of |g| with
+   {{init.}}{{MediaStreamTrackGenerator/[kind]}}.
+5. Return |g|.
+
+### Attributes
+<dl>
+<dt><dfn attribute for=MediaStreamTrackGenerator>writable</dfn></dt>
+<dd>Allows writing media frames the {{MediaStreamTrackGenerator}}, which is
+itself a {{MediaStreamTrack}}. If the {{MediaStreamTrackGenerator/[kind]}} attribute is `"audio"`,
+the stream will accept {{AudioFrame}} objects and fail with any other type. If
+{{MediaStreamTrackGenerator/[kind]}} is `"video"`, the stream will accept {{VideoFrame}} objects
+and fail with any other type. When a frame is written to {{MediaStreamTrackGenerator/writable}},
+its `close()` method is automatically invoked, so that its internal resources are no longer
+referenced from the Web.
+</dd>
+<dt><dfn attribute for=MediaStreamTrackProcessor>readableControl</dfn></dt>
+<dd>Allows reading control signals sent from any sinks connected to the
+{{MediaStreamTrackGenerator}}. Control signals are objects of type {{MediaStreamTrackSignal}}.
+</dd>
+</dl>
+
+## Stream control ## {#stream-control}
+<pre class="idl">
+dictionary MediaStreamTrackSignal {
+  required MediaStreamSignalType signalType;
+  [EnforceRange] unsigned long long fps;
```

Comment:
fps should be a double, and called frameRate, to fit with https://w3c.github.io/mediacapture-main/getusermedia.html#def-constraint-frameRate (support frame rates < 1, as well as the infamous 29.97).

Reply:
done